src/itwinai/slurm/slurm_script_builder.py

```diff
@@ -0,0 +1,331 @@
+
+        if not retain_file:
+            file_path.unlink()
+
+    def run_slurm_script_all_strategies(
+        self,
+        file_folder: Path = Path("slurm_scripts"),
+        retain_file: bool = True,
+        submit_slurm_job: bool = True,
+        strategies: List[str] = ["ddp", "horovod", "deepspeed"],
+    ):
+        """Runs the SLURM script with all the given strategies. Does the same as the
+        ``runall.sh`` script has been doing.
+        """
+        self.file_folder = file_folder
+        for strategy in strategies:
+            self.distributed_strategy = strategy
+
+            # Overriding job_name, std_out and err_out
+            self.slurm_script_configuration.job_name = self.generate_identifier()
+            std_out_path = Path("slurm_jobs") / (self.generate_identifier() + ".out")
+            err_out_path = Path("slurm_jobs") / (self.generate_identifier() + ".err")
+            self.slurm_script_configuration.std_out = std_out_path
+            self.slurm_script_configuration.err_out = err_out_path
+
+            self.process_slurm_script(
+                retain_file=retain_file, submit_slurm_job=submit_slurm_job
+            )
+
+    def run_scaling_test(
+        self,
+        file_folder: Path = Path("slurm_scripts"),
+        retain_file: bool = True,
+        submit_slurm_job: bool = True,
+        strategies: List[str] = ["ddp", "horovod", "deepspeed"],
+        num_nodes_list: List[int] = [1, 2, 4, 8],
+    ):
+        """Runs a scaling test, i.e. runs all the strategies with separate runs for each
+        distinct number of nodes.
+        """
+        for num_nodes in num_nodes_list:
+            self.slurm_script_configuration.num_nodes = num_nodes
+            self.run_slurm_script_all_strategies(
+                file_folder=file_folder,
+                retain_file=retain_file,
+                submit_slurm_job=submit_slurm_job,
+                strategies=strategies,
+            )
+
+
+def main():
+    parser = get_slurm_job_parser()
+    args = parser.parse_args()
+
+    slurm_script_configuration = SlurmScriptConfiguration(
+        job_name=args.job_name,
+        account=args.account,
+        time=args.time,
+        partition=args.partition,
+        std_out=args.std_out,
+        err_out=args.err_out,
+        num_nodes=args.num_nodes,
+        num_tasks_per_node=args.num_tasks_per_node,
+        gpus_per_node=args.gpus_per_node,
+        cpus_per_gpu=args.cpus_per_gpu,
+    )
+
+    submit_slurm_job = not args.no_submit_slurm_job
+    retain_file = not args.no_retain_file
+
+    slurm_script_builder = SlurmScriptBuilder(
+        slurm_script_configuration=slurm_script_configuration,
+        distributed_strategy=args.dist_strat,
+        debug=args.debug,
+        training_command=args.training_cmd,
+    )
+
+    slurm_script_builder.process_slurm_script(
+        retain_file=retain_file, submit_slurm_job=submit_slurm_job
+    )
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
it's fine to leave this here, but I was wondering whether you forgot to remove this. Maybe this could be converted in unit tests?